session.go

```diff
@@ -2254,29 +2254,46 @@ type traceWriter struct {
 	session *Session
 	w       io.Writer
 	mu      sync.Mutex
+
+	tracingInfoFetchAttempts int
+	tracingInfoFetchInterval time.Duration
 }
 
 // NewTraceWriter returns a simple Tracer implementation that outputs
 // the event log in a textual format.
 func NewTraceWriter(session *Session, w io.Writer) Tracer {
-	return &traceWriter{session: session, w: w}
+	return &traceWriter{session: session, w: w, tracingInfoFetchAttempts: 5, tracingInfoFetchInterval: 3 * time.Millisecond}
 }
 
 func (t *traceWriter) Trace(traceId []byte) {
 	var (
 		coordinator string
 		duration    int
 	)
-	iter := t.session.control.query(`SELECT coordinator, duration
+
+	fetchAttempts := 1
+	if t.tracingInfoFetchAttempts > 0 {
+		fetchAttempts = t.tracingInfoFetchAttempts
+	}
+
+	for i := 0; i < fetchAttempts; i++ {
+		iter := t.session.control.query(`SELECT coordinator, duration
```

Comment:
Scaning for activities should go first, because it signals when trace is completed, otherwise it could be the case that this information is not full as well.